Updated structure, new features, under a Jekyll template

DevPranjal opened the pull request:
Apologies for a large PR. Forgot to add `_site` into `.gitignore` for the first commit.

Preview: https://devpranjal.github.io/ML-InterviewQs/

Everything is now under a website! Since we do not have Github Actions currently, we will be able to setup the website under the dsgiitr domain only after the PR is merged.

This will also provide a base for further contributions, given that Hacktoberfest is getting closer.

DevPranjal committed:
fresh structure, new features, everything under a jekyll template

DevPranjal committed:
update gitignore to include jekyll generated files

AnirudhDagar commented on _data/rl..yml:
```diff
@@ -0,0 +1 @@
+# Put your questions here:
```
This line is fine but i think there is a typo in the name of the file, should be `rl.yml`

AnirudhDagar commented on _config.yml:
```diff
@@ -0,0 +1,59 @@
+# in the templates via {{ site.myvariable }}.
+
+title: ML Questions
+email: your-email@example.com
```
```suggestion
email: dsg@iitr.ac.in
```

AnirudhDagar commented on _config.yml:
```diff
@@ -0,0 +1,59 @@
+  ML Interview Questions
+# baseurl: "" # the subpath of your site, e.g. /blog
+url: "http://mlinterviews.com/" # the base hostname & protocol for your site, e.g. http://example.com
+twitter_username: jekyllrb
+github_username:  jekyll
```
Please update these.

AnirudhDagar reviewed (changes requested):
@DevPranjal I made a cursory pass on this PR. Thanks a lot for taking the pain to set up everything with the website. Using JustTheDocs is a nice choice IMO (love the rendered website).

One thing that I'd suggest is to not push anything that is generated using jekyll on the main branch. Instead, we should have a simple gh-action to generate and build, for example, the `_site` contents on every commit and PR at a different branch say "gh-pages" or something else. This is easy to do and should not be very tough since you have already done all the heavy lifting. Maybe take a look at [dsgiitr/website](https://github.com/dsgiitr/website/tree/gh-pages/docs) for some inspiration on how it should be set up. Then we add `_site` dir to `.gitignore` so that those changes are never pushed accidentally even after local build changes. 

I've also made a few very small comments, please take a look.

If you'd like, we can first merge this PR after you remove the generated files from the PR and then you can later send another PR for setting up the gh-actions on main as well as CI on PRs for previews etc. to deploy the website.

Overall this is looking in great shape. Great work!

AnirudhDagar commented:
Also, why is `_data` dir not named `data`? I believe anything starting with underscore here is to mark that the dir is automatically generated. Or maybe I'm wrong, please correct me in that case.

DevPranjal commented:
Thanks for the review!

I have addressed the queries below:
- `_site` was committed by mistake, I have added it to `.gitignore` and I'll also remove them from this PR
- Jekyll's Liquid template system can only access data files if they are stored in the `_data` directory precisely (as far as I know)
- I will look into other CI and build setups and make the changes in a new PR.

AnirudhDagar commented:
I think you forgot to push the commits @DevPranjal
